### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by Nicolas-Reyland**
I propose two small improvements :
 - using a generator instead of a list, when the list of converted to a set anyway
 - renaming of an unused variable to the underscore character

### CODE REVIEW

**Comment by Nicolas-Reyland** [comment]
I also re-wrote string formatting, using fstrings instead of the old-way %-formats and used the `.format` functions where f-strings were not suited.

File: findimports.py [hunk 1]
```diff
@@ -484,8 +487,8 @@ def parseFile(self, filename):
             module.unused_names = None
         dir = os.path.dirname(filename)
         module.imports = set(
-            [self.findModuleOfName(imp.name, imp.level, filename, dir)
-             for imp in module.imported_names])
+            self.findModuleOfName(imp.name, imp.level, filename, dir)
+            for imp in module.imported_names)
```
Comment (mgedmin) [inline findimports.py:491]: This could be a set comprehension, i.e. `{self.findModuleOfName(...) for ...}`.

File: findimports.py [hunk 2]
```diff
@@ -625,7 +628,7 @@ def listModules(self):
         """Return an alphabetical list of all modules."""
         modules = list(self.modules.items())
         modules.sort()
-        return [module for name, module in modules]
+        return [module for _, module in modules]
```
Comment (mgedmin) [inline findimports.py:631]: I prefer the expressiveness of `name, module` to remind the reader about the shape of the data structure, even if `name` is unused here.

Is there a linter that shows a warning about an unused name here?  If yes, can we silence it by using `_name`?  If not, I'd rather revert this bit.

**Comment by Nicolas-Reyland** [comment]
It seems there are errors because some lines are too long (80 chars instead of 79). I'll be looking into it if you're OK with it.

**Review by mgedmin — APPROVED** [review]
Thank you, I like it!

**Comment by mgedmin** [comment]
(Why on Earth am I running flake8 for every Python version, in addition to the separate flake8 job?)

File: findimports.py [hunk 3]
```diff
@@ -122,9 +122,12 @@ def __init__(self, name, filename, lineno, level):
         self.level = level
 
     def __repr__(self):
-        return '%s(%r, %r, %r, %r)' % (
-            self.__class__.__name__, self.name, self.filename, self.lineno,
-            self.level,
+        return "{class_name}('{name}', '{filename}', {lineno}, {level})".format(
```
Comment (mgedmin) [inline findimports.py]: flake8 says:
> findimports.py:125:80: E501 line too long (80 > 79 characters)

File: findimports.py [hunk 4]
```diff
@@ -781,24 +784,22 @@ def printDot(self):
...
-                nameDict[name] = id = "extmod%d" % n
-                print("  %s[label=\"%s\"];" % (id, name))
+                nameDict[name] = id = f"extmod{n}"
+                print(f"  {id}[label=\"{name}\"];")
         for modname, module in sorted(self.modules.items()):
             for other in sorted(module.imports):
                 if other in nameDict:
-                    print("  %s -> %s;" % (nameDict[module.modname],
-                                           nameDict[other]))
+                    print(f"  {nameDict[module.modname]} -> {nameDict[other]};")
```
Comment (mgedmin) [inline findimports.py]: flake8 says
> findimports.py:802:80: E501 line too long (80 > 79 characters)

**Review by mgedmin — COMMENTED** [review]
This fails CI.

I'm not opposed to increasing flake8 line length to something like black's default 88 in `setup.cfg`.  (I _think_ I have a setup.cfg with a [flake8] section here.)

File: findimports.py (hunk 2 above)
Comment (Nicolas-Reyland) [reply findimports.py:631]: No problem! I understand you point of view. Let's revert it.

**Review by mgedmin — APPROVED** [review]

File: findimports.py (hunk 3 above)
Comment (Nicolas-Reyland) [reply findimports.py]: I think I just fixed that

File: findimports.py (hunk 4 above)
Comment (Nicolas-Reyland) [reply findimports.py]: I think I just fixed that too, if I'm not mistaken

**Review by mgedmin — APPROVED** [review]

**Comment by mgedmin** [comment]
> (Why on Earth am I running flake8 for every Python version, in addition to the separate flake8 job?)

Historical reasons: I used Travis CI before, where the easiest path was to use the same build script for all Python versions; this was faithfully migrated to GitHub Actions later.  I think I added a separate flake8 GHA job much later, copying & pasting it to all 19 Python projects I maintain, without double-checking whether any of them already ran flake8 as part of the main build script.

Fixed in git master.

**Pull request merged**